scripts/reindex_articles.coffee

```diff
@@ -5,32 +5,115 @@ path = require 'path'
 Article = require '../src/api/apps/articles/model/index.js'
 ArticleModel = require '../src/api/models/article.coffee'
 search = require '../src/api/lib/elasticsearch'
-async = require 'async'
+asyncLib = require 'async'
 { cloneDeep } = require 'lodash'
 
-# Setup environment variables
 env = require 'node-env-file'
 switch process.env.NODE_ENV
   when 'test' then env path.resolve __dirname, '../.env.test'
   when 'production', 'staging' then ''
   else env path.resolve __dirname, '../.env'
 
-# Connect to database
-client = new MongoClient(process.env.MONGOHQ_URL)
-await client.connect()
-db = await client.db()
-articles = await db.collection('articles')
+console.log('Starting article reindexing process...')
+console.log('Environment:', process.env.NODE_ENV || 'development')
+console.log('MongoDB URL:', process.env.MONGOHQ_URL ? 'Configured' : 'NOT CONFIGURED')
 
-articles.find({}).toArray (err, articles) ->
-  console.log(err) if err
-  console.log(articles.length)
-  async.mapSeries(articles, indexWorker, (err, results) =>
-    console.log('Completed indexing ' + results.length + ' articles.')
-  )
+# Log Elasticsearch configuration
+console.log('Elasticsearch Configuration:')
+console.log('  - URL:', process.env.ELASTICSEARCH_URL || process.env.OPENSEARCH_URL || 'NOT CONFIGURED')
+console.log('  - Index Name:', search.index)
+console.log('  - Index Suffix:', process.env.ELASTICSEARCH_INDEX_SUFFIX || 'production')
+console.log('  - Using OpenSearch:', search.isOpenSearch ? 'Yes' : 'No')
+
+main = ->
+  startTime = Date.now()
+  console.log('Connecting to MongoDB...')
+
+  client = new MongoClient(process.env.MONGOHQ_URL)
+  client.connect()
+    .then (client) ->
+      console.log('✓ Connected to MongoDB successfully')
+      db = client.db()
+      console.log('✓ Database connection established')
+      return db.collection('articles')
+    .then (articlesCollection) ->
+      console.log('✓ Articles collection accessed')
+      console.log('Fetching all articles from database (newest first)...')
+      articlesCollection.find({}).sort({_id: -1}).toArray()
+    .then (articles) ->
+      console.log('✓ Found ' + articles.length + ' articles to index')
+      console.log('Starting indexing process...')
+
+      new Promise (resolve, reject) ->
+        processedCount = 0
+        errorCount = 0
+
+        asyncLib.mapSeries articles, (article, cb) ->
+          processedCount++
+          console.log("[#{processedCount}/#{articles.length}] Indexing article: #{article._id} (#{article.title || 'No title'})")
+
+          indexWorker article, (err) ->
```

Comment:
Bit confused, are we duplicating the same logging below as well as the `indexWorker` function? Seems like we will log successes and errors for each run, twice?